### PR TITLE
add a simple_destructor for file objects

### PR DESCRIPTION
**Pull request by toshok**
tested in gdb by setting a breakpoint in fileDestructor and running this test:

```python
import gc

def foo():
  f = open("/dev/null")

foo()
gc.collect()
```

### CODE REVIEW

**Comment by kmod** [comment]
Can we get a test?  like the one that you put in the comment -- or something like opening 10k files (and maybe calling gc.collect every 100 of them).

Also, formatting :)

**Comment by toshok** [comment]
Yeah, I clearly need a pre-commit hook to check formatting :)

File: test/tests/file_destructor.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+try:
```
Comment (kmod) [inline test/tests/file_destructor.py]: Couple minor things as we wait for the travis-ci build to rerun:
- It's ok to count on the `gc` module being available
- let's use 4 spaces for indenting -- we try to follow [PEP8](https://www.python.org/dev/peps/pep-0008/#id12) for our Python code
- we could try calling `gc.collect()` every ~100 loop iterations to reduce that overhead

**Comment by kmod** [comment]
Cool :)

**Pull request merged**